gyb_syntax_support/TypeNodes.py

```diff
@@ -123,6 +123,14 @@
              Child('Ellipsis', kind='EllipsisToken')
          ]),
 
+    # pack-reference-type -> 'each' type
+    Node('PackReferenceType', name_for_diagnostics='pack reference', kind='Type',
```

Comment:
This node is identical to the `ConstrainedSugarType`. What do you think about using the same node type for it instead of introducing a new node? I'm open to naming suggestions.

Reply:
Hmm. You're right that the syntactic structure is the same (though it's parsed slightly differently), but conceptually a pack element type is very different from a constrained sugar type. Do you think that most tooling operating on these type syntax nodes would want to treat pack elements the same as constraint types? The only reason why I added a new syntax node is because I added a new `TypeRepr` on the Swift side to make type resolution easier, since the resolution logic is completely different.